### PR TITLE
IGAPP-973: Mark external links

### PR DESCRIPTION
**Pull request by LeandraH**
This pull request belongs to the issue https://issues.tuerantuer.org/browse/IGAPP-973.

To test on web: Go to http://localhost:9000/testumgebung/de/test-f%C3%BCr-externe-links, look at the links
To test on native: Go to Testumgebung, in German, click on Test für externe Links and look at the links.

### CODE REVIEW

File: web/src/components/__tests__/RemoteContent.spec.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ describe('RemoteContent', () => {
 
   it('should not trigger on internalLinkClick for external links', () => {
     const href = `https://some.external/link`
-    const html = `<a href=${href}>Test Anchor</a>`
+    const html = `<a href=${href} class="external-link">Test Anchor</a>`
```
[the remark this answers is not in the document]
Comment (LeandraH) [reply web/src/components/__tests__/RemoteContent.spec.tsx]: True, thanks!

File: release-notes/unreleased/IGAPP-973-mark-external-links.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+issue_key: IGAPP-973
+show_in_stores: true
+platforms:
+  - web
+  - android
+  - ios
+en: Mark external links with an icon
+de: Externe Links mit einem Icon markieren
```
Comment (f1sh1918) [inline release-notes/unreleased/IGAPP-973-mark-external-links.yml:8]:
```suggestion
de: Externe Links werden mit einem Icon markiert
```

**Review by f1sh1918 — COMMENTED** [review]
I have some rendering problems which chrome, check screenshot. This can be fixed by using `display:flex` for link and `after-Element` but this may cause other errors, so pls check before 

image

**Comment by LeandraH** [comment]
Okay, I tested it on OS X Firefox, Chrome, Safari, and iOS and Android simulations, and it all looks good there. Would love for someone with Windows to test there a little :)
The weird underline didn't actually happen in native but I think it's better to still give it the display flex, just in case it changes some behavior down the line.

**Review by steffenkleinle — COMMENTED** [review]
Nicely done!

Do you think it would be possible to also implement this for the NativeHtml component in native? Otherwise this does not work in the list content, see screenshot:

File: native/src/utils/renderHtml.ts [hunk 3]
```diff
@@ -222,6 +222,22 @@ const renderHtml = (
...
         overflow-x: auto;
       }
+
+      .link-external {
+        display: flex;
+        align-items: center;
+      }
+
+      .link-external::after {
+        content: url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='1114.4419786503477 1814.3063537312773 18 18' width='18' height='18' style='-webkit-print-color-adjust:exact'><path d='M1130.442 1830.306h-14v-14h7v-2h-7a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7Zm-5-16v2h3.59l-9.83 9.83 1.41 1.41 9.83-9.83v3.59h2v-7h-7Z' fill='rgb(11, 87, 208)'/></svg>");
```
[the remark this answers is not in the document]
Comment (LeandraH) [reply native/src/utils/renderHtml.ts:232]: See the comment under it ;) I haven't been able to get it to work :(

**Comment by LeandraH** [comment]
> Nicely done!
> 
> Do you think it would be possible to also implement this for the NativeHtml component in native? Otherwise this does not work in the list content, see screenshot: 

I'll give it a try :)

File: web/src/components/RemoteContent.tsx [hunk 4]
```diff
@@ -12,7 +13,7 @@ const SandBox = styled.div<{ centered: boolean; smallText: boolean }>`
 
   ${props => (props.centered ? 'text-align: center;' : '')}
   ${props => (props.centered ? 'list-style-position: inside;' : '')}
-  & img {
+  img {
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply web/src/components/RemoteContent.tsx:16]: Thanks for explaining, sounds good!

**Comment by LeandraH** [comment]
Looks like `react-native-render-html` can't handle a `<path>` element so getting the icon into the `NativeHtml` component seems to be a no in the foreseeable future :( Looking at the speed difference, I don't think it's worth it to replace `react-native-render-html` for such a small feature.

**Comment by f1sh1918** [comment]
> Looks like `react-native-render-html` can't handle a `<path>` element so getting the icon into the `NativeHtml` component seems to be a no in the foreseeable future :( Looking at the speed difference, I don't think it's worth it to replace `react-native-render-html` for such a small feature.

Indeed the only solution i see is to create a logic to inject `` or `<svg>` tags in this particular cases with the icon (before `</a` closes. but thats very hacky and can lead to errors. Not sure if thats worth to do

**Comment by steffenkleinle** [comment]
Is it somehow possible to use png or some other image format?

**Review by steffenkleinle — APPROVED** [review]
If it is currently not possible to use svgs in react-native-render-html could you perhaps open an issue there and in our jira to fix this later?

File: native/src/components/NativeHtml.tsx [hunk 5]
```diff
@@ -85,6 +100,14 @@ const NativeHtml = React.memo(({ content, cacheDictionary, language }: NativeHtm
         textAlign: contentAlignment(language),
         direction: config.hasRTLScript(language) ? 'rtl' : 'ltr',
       }}
+      customHTMLElementModels={{
+        svg: defaultHTMLElementModels.svg.extend({
+          contentModel: HTMLContentModel.mixed,
+        }),
+      }}
+      renderers={{
+        svg: () => <SimpleImage source={ExternalIcon} />,
```
Comment (f1sh1918) [inline native/src/components/NativeHtml.tsx:109]:
```suggestion
        svg: () => <SimpleImage style={{ marginLeft: 10 }} source={ExternalIcon} />,
```

**Review by f1sh1918 — APPROVED** [review]
Really nice work :) thx for the effort you put in. Works nice.
Just added the little missing piece, maybe it way too obvious :)

**Review by steffenkleinle — COMMENTED** [review]
Tested on android, thanks for investing the time to find a solution!

Maybe you could write some tests for this :)

One more thing: For me the icons look quite big and I think it could be a little smaller. Did you follow a design? Since I am not an UI/UX expert you may also just ignore this if you think its better this way or this was like this in a design.

File: native/src/components/NativeHtml.tsx (hunk 5 above)
Comment (LeandraH) [reply native/src/components/NativeHtml.tsx:109]: True, thanks! Hab ich auf dem Schlauch gestanden...

**Review by steffenkleinle — APPROVED** [review]

**Pull request merged**